files/Aboot/boot0.j2

```diff
@@ -317,106 +444,150 @@ platform_specific() {
-        echo "hwaddr_ma1=$(cat /sys/class/net/ma1/address)" >> /tmp/append
+        cmdline_add "hwaddr_ma1=$(cat /sys/class/net/ma1/address)"
     elif [ -f /sys/class/net/eth0/address ]; then
-        echo "hwaddr_ma1=$(cat /sys/class/net/eth0/address)" >> /tmp/append
+        cmdline_add "hwaddr_ma1=$(cat /sys/class/net/eth0/address)"
     else
         err "Management port not found."
     fi
 
-    # use extra parameters from kernel-params hook if the file exists
-    if [ -f "$target_path/$kernel_params" ]; then
-        cat "$target_path/$kernel_params" >> /tmp/append
+    # Obtain root partition uuid
+    local rootdev="$(mount | grep -E "$flash_re" | cut -f1 -d' ')"
+    local rootfstype="$(mount | grep -E "$flash_re" | cut -f5 -d' ')"
+    local rootuuid="$(get_uuid_for $rootdev)"
+    if [ -z "$rootuuid" ] || [ "$rootfstype" = "vfat" ] ; then
+        cmdline_add "root=$rootdev"
+    else
+        cmdline_add "root=UUID=$rootuuid"
     fi
+}
 
-    # setting root partition if not overridden by kernel-params
-    if ! grep -q "root=" /tmp/append; then
-        rootdev="$(mount | grep -E "$flash_re" | cut -f1 -d' ')"
-        rootfstype="$(mount | grep -E "$flash_re" | cut -f5 -d' ')"
-        rootuuid="$(get_uuid_for $rootdev)"
-        if [ -z "$rootuuid" ] || [ "$rootfstype" = "vfat" ] ; then
-            echo "root=$rootdev" >> /tmp/append
-        else
-            echo "root=UUID=$rootuuid" >> /tmp/append
-        fi
+write_default_cmdline() {
+    local delimiter="cmdline-aboot-end"
+
+    cmdline_clear
+
+    if $in_aboot; then
+        # generate the default kernel parameters for the platform
+        cat /etc/cmdline | sed "/^\(${bootconfigvars// /\|}\|crashkernel\|loglevel\|ignore_loglevel\)\(\$\|=\)/d;/^\$/d" | cmdline_append
+    elif grep -q "$delimiter" /proc/cmdline; then
+        # we are on a recent sonic image using delimiter. extracting the part of the
+        # cmdline coming from aboot is trivial.
+        cat /proc/cmdline | sed -E "s/^(.*) $delimiter .*$/\1/" | tr ' ' '\n' | cmdline_append
+    else
+        # we are either on SONiC or EOS and the commandline doesn't have a delimiter
+        # for the Aboot part. Take an educated guess at a right delimiter.
+        # Subject to breakage if EOS or SONiC cmdline change.
+        cat /proc/cmdline | sed -E 's/^(.*) rw .*$/\1/' | tr ' ' '\n' | cmdline_append
     fi
 
-    mkdir -p "$image_path"
-    cat /tmp/append > $cmdline_image
-    [ -s ${target_path}/machine.conf ] || write_machine_config
-    sync
+    cmdline_add "$delimiter"
+}
+
+write_common_configs() {
+    write_default_cmdline
+    write_platform_specific_cmdline
+    write_image_specific_cmdline
+    write_machine_config
+}
+
+write_secureboot_configs() {
+    write_common_configs
+    cmdline_add "loop=${installer_image_path/mnt\/flash\//}"
+    cmdline_add loopfstype=squashfs
+    cmdline_add "loopoffset=$(unzip -qqf "$installer_image_path" fs.squashfs)"
+    cmdline_add docker_inram=on
```

Comment:
secureboot should be independent on docker_inram. Since we extract it each cold reboot, there is no risk to be tampered.

Reply:
@lguohan please check if it is valid that secureboot should be independent on docker_inram?

Reply:
I will go ahead and make `secure_boot_enable=y` force `docker_inram=on` in `union-mount.j2` we can always revisit later.